net-utils/src/ip_echo_client.rs

```diff
@@ -177,15 +178,21 @@ pub(crate) async fn do_verify_reachable_ports(
                 ok = false;
             }
         }
-        thread_handle.join().expect("Thread should exit cleanly")
+        thread_handle.await.expect("Thread should exit cleanly")
     }
 
-    if !ok {
-        // No retries for TCP, abort on any failure
-        return false;
-    }
+    ok
+}
 
-    // now check UDP ports
+/// Checks if all of the provided UDP ports are reachable by the machine at
+/// `ip_echo_server_addr`. Tests must complete within timeout provided.
+/// A given amount of retries will be made to accommodate packet loss.
+pub(crate) async fn verify_reachable_udp_ports(
+    ip_echo_server_addr: SocketAddr,
+    udp_sockets: &[&UdpSocket],
+    timeout: Duration,
+    retry_count: usize,
+) -> bool {
```

Comment:
I struggle to understand why it is bool instead of `Result`. 

And why do we need `ok`. Even if one of the ports is unavailable, we will exit validator (see bootstrap line 607).
So there is no reason to continue when we found first unavailable port.

Reply:
i see that there is a comment that we should go over all the ports regardless of errors. But what will happen if thread will not finish cleanly?